### PR TITLE
Make action work on linux

### PR DESCRIPTION
**Pull request by einarmo**
Use osslsigncode to sign binaries on linux as well.

This was slightly painful. The version of openssl installed by default requires specifying the "legacy" flag to work with RC2 certificates, the easiest way I found was to pass the certificate through openssl. This depends on ubuntu version as well, openssl 1.x on 20 works, 3.x on 22 requires this workaround.

### CODE REVIEW

File: sign.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        -export -out ./cognite_code_signing_2.pfx \
+        -passin "pass:$CERTIFICATE_PASSWORD" -passout "pass:$CERTIFICATE_PASSWORD"
+    mv ./cognite_code_signing_2.pfx ./cognite_code_signing.pfx
+    rm ./cognite_code_signing.pem
+fi
+
+recurse=false
+file_path="$1"
+
+for var in "$@"; do
```
Comment (bisera-cognite) [inline sign.sh:27]: Do you need to iterate the array and keep track of the recurse value? Would something like this work `[[ $@ =~ "-Recurse" ]]` instead?

File: sign.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            -t "http://timestamp.digicert.com" \
+            -in "$f" -out "$f.signed"
+        mv "$f.signed" "$f"
+    done
+else
+    echo "Sign a single binary"
+    osslsigncode sign -pkcs12 ./cognite_code_signing.pfx -pass "$CERTIFICATE_PASSWORD" \
+        -t "http://timestamp.digicert.com" \
+        -in "$file_path" -out "$file_path.signed"
+    mv "$file_path.signed" "$file_path"
```
Comment (bisera-cognite) [inline sign.sh:46]: Since it's duplicated code maybe it would be more readable to move it to a function?

File: sign.sh (hunk 1 above)
Comment (einarmo) [reply sign.sh:27]: Sure, my shellscript is rather bad...

Comment (einarmo) [reply sign.sh:27]: I think the current solution is really the correct way to do it in shellscript. If there were more arguments we could use a `case` statement, but I think this is fine.

**Review by bisera-cognite — APPROVED** [review]

File: sign.sh (hunk 1 above)
Comment (bisera-cognite) [reply sign.sh:27]: Yes, I agree. PR approved :)

**Pull request merged**